src/library-authoring/common/data/utils.js

```diff
@@ -61,3 +61,128 @@ export const unpackLibraryKey = (key) => {
 export const truncateErrorMessage = (errorMessage) => (
   errorMessage.length > 255 ? `${errorMessage.substring(0, 255)}...` : errorMessage
 );
+
+
+const tagMap = [
+  ['attribution', 'BY'],
+  ['nonCommercial', 'NC'],
+  ['noDerivatives', 'ND'],
+  ['shareAlike', 'SA'],
+];
+
+// Some flags for Creative Commons are mutually exclusive.
+const exclusiveMap = {
+  noDerivatives: 'shareAlike',
+  shareAlike: 'noDerivatives',
+};
+
+const defaultFlags = {
+  attribution: true,
+  nonCommercial: false,
+  noDerivatives: false,
+  shareAlike: false,
+};
+
+const optionsMatchers = {
+  // Always required. Don't give the chance for junk data to break this.
+  attribution: /.*/,
+  nonCommercial: /:NC(:|$)/,
+  noDerivatives: /:ND(:|$)/,
+  shareAlike: /:SA(:|$)/,
+};
+
+/**
+ * A number, or a string containing a number.
+ * @typedef {{noDerivatives: boolean, nonCommercial: boolean, attribution: boolean, shareAlike: boolean}} CommonsOptions
+ */
```

Comment:
This docstring looks like it's misplaced, or just obsolete.

Reply:
@arbrandes This should have only contained the second line-- I think I copied the first one by mistake from one of the functions I was writing docs for at the time.

The second line is a JSDoc type declaration, which allows me to reference the type in later function declarations and expect my and others' IDEs to pick them up while conforming to @bradenmacdonald 's request we add JSDoc comments to things (especially components, but I figured I'd add them to functions here since some of them could use the extra explanations and it's good practice.)

Fixed.